pipelines/pipeline_katello_34_to_35_upgrade.yml

```diff
@@ -110,6 +126,69 @@
     - katello_repositories
     - foreman_installer
 
+# Check that we're still good on 3.5 with Puppet 4
+- hosts: pipeline-katello-3.4-3.5-centos7
+  become: true
+  vars:
+    bats_tests:
+      - fb-test-katello.bats
+      - fb-destroy-organization.bats
+      - fb-content-katello.bats
+      - fb-proxy.bats
+      - fb-finish.bats
+  roles:
+    - bats
+
+# Upgrade to Puppet 5
+# http://projects.theforeman.org/projects/foreman/wiki/Upgrading_from_Puppet_4_to_5#Prepare-And-Run-The-Installer
+- hosts:
+  - pipeline-katello-3.4-3.5-centos7
+  - pipeline-proxy-3.4-3.5-centos7
+  become: true
+  vars:
+    puppet_repositories_version: 5
+  roles:
+    - puppet_repositories
+
+- hosts:
+  - pipeline-katello-3.4-3.5-centos7
+  - pipeline-proxy-3.4-3.5-centos7
+  become: true
+  tasks:
+    - name: 'Update packages'
+      yum:
+        name: '*'
+        state: latest
+    - name: 'Reinstall puppet-agent-oauth'
+      command: yum reinstall -y puppet-agent-oauth
+
+- hosts: pipeline-katello-3.4-3.5-centos7
+  become: true
+  vars:
+    foreman_installer_scenario: katello
+    foreman_installer_options_internal_use_only:
+      - "--disable-system-checks"
+      - "--reset-puppet-server-jvm-extra-args"
+      - "--reset-puppet-server-puppetserver-version"
+      - "--reset-puppet-server-puppetserver-metrics"
+      - "--reset-puppet-server-puppetserver-experimental"
+  roles:
+    - foreman_installer
```

Comment:
Is this part just to upgrade puppet?

Reply:
No, this part resets some Foreman variables about the puppet server. I think it discovers the Puppet version and stuff if they're not set. The actual puppet upgrade happens on [line 160](https://github.com/theforeman/forklift/pull/710/files#diff-8b9d90c21ca1de611ed04bd30b9685d1R160), so by this point, if you ran `puppet --version`, it'd say `5.5.1`.

Reply:
I'm getting these steps from [this wiki](http://projects.theforeman.org/projects/foreman/wiki/Upgrading_from_Puppet_4_to_5#Prepare-And-Run-The-Installer).